[Bug] 【doc】Table name[MyTable] cannot contain upper case

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.


### Paimon version

0.4

### Compute Engine

flink

### Minimal reproduce step

CREATE CATALOG my_hive WITH (
>     'type' = 'paimon',
>     'metastore' = 'hive',
>     'uri' = 'thrift://***:9083',
>     'warehouse' = 'hdfs://***/user/zym/paimon'
> );

Flink SQL> USE CATALOG my_hive;
[INFO] Execute statement succeed.

CREATE TABLE MyTable (
>     user_id BIGINT,
>     item_id BIGINT,
>     behavior STRING,
>     dt STRING,
>     hh STRING,
>     PRIMARY KEY (dt, hh, user_id) NOT ENFORCED
> );
[ERROR] Could not execute SQL statement. Reason:
java.lang.IllegalStateException: Table name[MyTable] cannot contain upper case

### What doesn't meet your expectations?

hive catalog org.apache.paimon.hive.HiveCatalog#checkIdentifierUpperCase  ;

do we need to change the table name "MyTable" to lowercase "my_table"?

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/docs/content/how-to/creating-catalogs.md
+++ b/docs/content/how-to/creating-catalogs.md
@@ -80,6 +80,8 @@ USE paimon.default;
 
 By using Paimon Hive catalog, changes to the catalog will directly affect the corresponding Hive metastore. Tables created in such catalog can also be accessed directly from Hive.
 
+To use Hive catalog, Database name、Table name and Field names should be lower case.
+
 {{< tabs "hive-metastore-example" >}}
 
 {{< tab "Flink" >}}

--- a/docs/content/how-to/creating-catalogs.md
+++ b/docs/content/how-to/creating-catalogs.md
@@ -80,7 +80,7 @@ USE paimon.default;
 
 By using Paimon Hive catalog, changes to the catalog will directly affect the corresponding Hive metastore. Tables created in such catalog can also be accessed directly from Hive.
 
-To use Hive catalog, Database name、Table name and Field names should be lower case.
+To use Hive catalog, Database name, Table name and Field names should be lower case.
 
 {{< tabs "hive-metastore-example" >}}
 

--- a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
+++ b/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
@@ -335,8 +335,6 @@ public class HiveCatalog extends AbstractCatalog {
             // first commit changes to underlying files
             TableSchema schema = schemaManager.commitChanges(changes);
 
-            // flink can alter table column since version 1.17
-            checkFieldNamesUpperCase(schema.fieldNames());
             try {
                 // sync to hive hms
                 Table table =

--- a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
+++ b/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
@@ -397,14 +397,14 @@ public class HiveCatalog extends AbstractCatalog {
 
     private void checkFieldNamesUpperCase(List<String> fieldNames) {
         List<String> illegalFieldNames =
-            fieldNames.stream()
-                .filter(f -> !f.equals(f.toLowerCase()))
-                .collect(Collectors.toList());
+                fieldNames.stream()
+                        .filter(f -> !f.equals(f.toLowerCase()))
+                        .collect(Collectors.toList());
         checkState(
-            illegalFieldNames.isEmpty(),
-            String.format(
-                "Field names %s cannot contain upper case in hive catalog",
-                illegalFieldNames));
+                illegalFieldNames.isEmpty(),
+                String.format(
+                        "Field names %s cannot contain upper case in hive catalog",
+                        illegalFieldNames));
     }
 
     private Database convertToDatabase(String name) {

--- a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
+++ b/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
@@ -305,6 +305,7 @@ public class HiveCatalog extends AbstractCatalog {
         }
 
         try {
+            checkIdentifierUpperCase(toTable);
             String fromDB = fromTable.getDatabaseName();
             String fromTableName = fromTable.getObjectName();
             Table table = client.getTable(fromDB, fromTableName);
@@ -334,6 +335,8 @@ public class HiveCatalog extends AbstractCatalog {
             // first commit changes to underlying files
             TableSchema schema = schemaManager.commitChanges(changes);
 
+            // flink can alter table column since version 1.17
+            checkFieldNamesUpperCase(schema.fieldNames());
             try {
                 // sync to hive hms
                 Table table =
@@ -368,12 +371,13 @@ public class HiveCatalog extends AbstractCatalog {
         checkState(
                 identifier.getDatabaseName().equals(identifier.getDatabaseName().toLowerCase()),
                 String.format(
-                        "Database name[%s] cannot contain upper case",
+                        "Database name[%s] cannot contain upper case in hive catalog",
                         identifier.getDatabaseName()));
         checkState(
                 identifier.getObjectName().equals(identifier.getObjectName().toLowerCase()),
                 String.format(
-                        "Table name[%s] cannot contain upper case", identifier.getObjectName()));
+                        "Table name[%s] cannot contain upper case in hive catalog",
+                        identifier.getObjectName()));
     }
 
     private void checkFieldNamesUpperCase(List<String> fieldNames) {
@@ -383,7 +387,9 @@ public class HiveCatalog extends AbstractCatalog {
                         .collect(Collectors.toList());
         checkState(
                 illegalFieldNames.isEmpty(),
-                String.format("Field names %s cannot contain upper case", illegalFieldNames));
+                String.format(
+                        "Field names %s cannot contain upper case in hive catalog",
+                        illegalFieldNames));
     }
 
     private Database convertToDatabase(String name) {

--- a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
+++ b/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java
@@ -330,6 +330,7 @@ public class HiveCatalog extends AbstractCatalog {
             }
         }
 
+        checkFieldNamesUpperCaseInSchemaChange(changes);
         try {
             final SchemaManager schemaManager = schemaManager(identifier);
             // first commit changes to underlying files
@@ -378,16 +379,32 @@ public class HiveCatalog extends AbstractCatalog {
                         identifier.getObjectName()));
     }
 
+    private void checkFieldNamesUpperCaseInSchemaChange(List<SchemaChange> changes) {
+        List<String> fieldNames = new ArrayList<>();
+        for (SchemaChange change : changes) {
+            if (change instanceof SchemaChange.AddColumn) {
+                SchemaChange.AddColumn addColumn = (SchemaChange.AddColumn) change;
+                fieldNames.add(addColumn.fieldName());
+            } else if (change instanceof SchemaChange.RenameColumn) {
+                SchemaChange.RenameColumn rename = (SchemaChange.RenameColumn) change;
+                fieldNames.add(rename.newName());
+            } else {
+                // do nothing
+            }
+        }
+        checkFieldNamesUpperCase(fieldNames);
+    }
+
     private void checkFieldNamesUpperCase(List<String> fieldNames) {
         List<String> illegalFieldNames =
-                fieldNames.stream()
-                        .filter(f -> !f.equals(f.toLowerCase()))
-                        .collect(Collectors.toList());
+            fieldNames.stream()
+                .filter(f -> !f.equals(f.toLowerCase()))
+                .collect(Collectors.toList());
         checkState(
-                illegalFieldNames.isEmpty(),
-                String.format(
-                        "Field names %s cannot contain upper case in hive catalog",
-                        illegalFieldNames));
+            illegalFieldNames.isEmpty(),
+            String.format(
+                "Field names %s cannot contain upper case in hive catalog",
+                illegalFieldNames));
     }
 
     private Database convertToDatabase(String name) {

--- a/paimon-hive/paimon-hive-catalog/src/test/java/org/apache/paimon/hive/HiveCatalogTest.java
+++ b/paimon-hive/paimon-hive-catalog/src/test/java/org/apache/paimon/hive/HiveCatalogTest.java
@@ -72,7 +72,8 @@ public class HiveCatalogTest extends CatalogTestBase {
                                         DEFAULT_TABLE_SCHEMA,
                                         false))
                 .hasRootCauseInstanceOf(IllegalStateException.class)
-                .hasRootCauseMessage("Database name[TEST_DB] cannot contain upper case");
+                .hasRootCauseMessage(
+                        "Database name[TEST_DB] cannot contain upper case in hive catalog");
 
         assertThatThrownBy(
                         () ->
@@ -81,6 +82,7 @@ public class HiveCatalogTest extends CatalogTestBase {
                                         DEFAULT_TABLE_SCHEMA,
                                         false))
                 .hasRootCauseInstanceOf(IllegalStateException.class)
-                .hasRootCauseMessage("Table name[NEW_TABLE] cannot contain upper case");
+                .hasRootCauseMessage(
+                        "Table name[NEW_TABLE] cannot contain upper case in hive catalog");
     }
 }

--- a/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/HiveCatalogITCaseBase.java
+++ b/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/HiveCatalogITCaseBase.java
@@ -512,7 +512,8 @@ public abstract class HiveCatalogITCaseBase {
                                                 "CREATE TABLE T ( a INT, b STRING ) WITH ( 'file.format' = 'avro' )")
                                         .await())
                 .hasRootCauseMessage(
-                        String.format("Table name[%s] cannot contain upper case", "T"));
+                        String.format(
+                                "Table name[%s] cannot contain upper case in hive catalog", "T"));
 
         assertThatThrownBy(
                         () ->
@@ -520,7 +521,9 @@ public abstract class HiveCatalogITCaseBase {
                                                 "CREATE TABLE t (A INT, b STRING, C STRING) WITH ( 'file.format' = 'avro')")
                                         .await())
                 .hasRootCauseMessage(
-                        String.format("Field names %s cannot contain upper case", "[A, C]"));
+                        String.format(
+                                "Field names %s cannot contain upper case in hive catalog",
+                                "[A, C]"));
     }
 
     @Test

--- a/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/HiveCatalogITCaseBase.java
+++ b/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/HiveCatalogITCaseBase.java
@@ -439,6 +439,11 @@ public abstract class HiveCatalogITCaseBase {
                 .hasMessage(
                         "Could not execute ALTER TABLE my_hive.test_db.t1 RENAME TO my_hive.test_db.t2");
 
+        // the target table name has upper case.
+        assertThatThrownBy(() -> tEnv.executeSql("ALTER TABLE t1 RENAME TO T1"))
+                .hasMessage(
+                        "Could not execute ALTER TABLE my_hive.test_db.t1 RENAME TO my_hive.test_db.T1");
+
         tEnv.executeSql("ALTER TABLE t1 RENAME TO t3").await();
         List<String> tables = hiveShell.executeQuery("SHOW TABLES");
         Assert.assertTrue(tables.contains("t3"));

